File: frontend/src/modules/stash/application/use-cases/readStash.ts
import { StashReader } from "@modules/stash/application/contracts/StashReader";
import { StashEntry } from "@modules/stash/domain/stashEntry";

export async function readStash(
  stashReader: StashReader,
  repositoryPath: string
): Promise<StashEntry[]> {
  return stashReader.readStash(repositoryPath);
}

File: backend/src/app/stash/read_stash.rs
use std::path::Path;

use crate::app::stash::contracts::StashReader;
use crate::domain::errors::AppError;
use crate::domain::stash_entry::StashEntry;

pub fn read_stash(
    stash_reader: &dyn StashReader,
    repository_path: &Path,
) -> Result<Vec<StashEntry>, AppError> {
    stash_reader.read_stash(repository_path)
}
